src/servers/http_server.cc

```diff
@@ -379,6 +381,13 @@ ReadDataFromJsonHelper(
         default:
           break;
       }
+      // Check if writing to 'serialized' is overrunning the expected byte_size
+      uint64_t bytes_read = *counter * TRITONSERVER_DataTypeByteSize(dtype);
```

Comment:
You don't need to compute the byte size on every element, in fact, you can just keep track of the expected total counter (expected number of elements) and compare with `*counter` directly, as `byte_size` is simply computed by `element_cnt * TRITONSERVER_DataTypeByteSize(dtype)`

From this perspective, note that the expected counter is slightly different from `TRITONSERVER_TYPE_BYTES`, which is just the byte size, so you need to condition to set it properly before calling into `ReadDataFromJson()`

Reply:
I see. Script is modified, please review.